hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/TwoToThreeUpgradeHandler.java

```diff
@@ -47,8 +50,13 @@
     tablePropsToAdd.put(HoodieTableConfig.HIVE_STYLE_PARTITIONING_ENABLE, config.getStringOrDefault(HoodieTableConfig.HIVE_STYLE_PARTITIONING_ENABLE));
     String keyGenClassName = Option.ofNullable(config.getString(HoodieTableConfig.KEY_GENERATOR_CLASS_NAME))
         .orElse(config.getString(HoodieWriteConfig.KEYGENERATOR_CLASS_NAME));
+    if (keyGenClassName == null && config.getEngineType() == EngineType.SPARK) {
```

Comment:
So other engines have other default key gen is it ?

Reply:
For Flink and Java, I don't see the default key generator class being set.  We can keep that for now.

Reply:
sg